HPCC-13391 Set default for row tag

miguelvazq opened the pull request:
When user try to spray XML file without Row Tag (with empty Row Tag), then the file will sprayed as a delimited (CSV like) file based on the encoding (UTF-xx). Will populate with a value as previously.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

Commit:
HPCC-13391 Set default for row tag

When user try to spray XML file without Row Tag (with empty Row Tag), then the file will sprayed as a delimited (CSV like) file based on the encoding (UTF-xx). Will populate with a value as previously.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

miguelvazq commented:
@GordonSmith Please review.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 05986d5bd972cf2f1c0d480d1e70ba228f77a4d7
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.407 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |722 |722 |0 |
|test (thor) |617 |617 |0 |
|test (roxie) |746 |746 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith reviewed (approved):
Looks fine

GordonSmith commented on esp/src/eclwatch/LZBrowseWidget.js:
```diff
@@ -800,7 +800,7 @@ define([
                         targetName: item.displayName,
                         targetRecordLength: "",
-                        targetRowTag: "",
+                        targetRowTag: context.i18n.tag,
```
Should the default tag be translated?

richardkchapman replied:
I wouldn't have thought so

miguelvazq replied:
I agree. I will update.

miguelvazq commented:
@GordonSmith Updated.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 5d6e40838d50a02147c3a806c826dd14d5eea24c
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 98.533 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |722 |722 |0 |
|test (thor) |617 |617 |0 |
|test (roxie) |746 |746 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith reviewed (approved):
Looks fine - but is there code missing now (or is it on a diff / previous PR)?

miguelvazq commented:
@GordonSmith In my originally commit I reverted to the original rowtag field value as initially created. I have now added some validation logic to disable spray button. Previously the spray was allowed regardless of field being empty or not.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 4dc3b3780e965555143f88cf2be97cbf09bc2f6f
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.127 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |722 |722 |0 |
|test (thor) |617 |617 |0 |
|test (roxie) |746 |746 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith reviewed (approved):
Looks good.

miguelvazq commented:
@richardkchapman This is ready to merge.

richardkchapman commented on esp/src/eclwatch/LZBrowseWidget.js:
```diff
@@ -800,7 +812,7 @@ define([
                         targetName: item.displayName,
                         targetRecordLength: "",
-                        targetRowTag: "",
+                        targetRowTag: "tag",
```
Is this right, or should it say Row?

GordonSmith replied:
Agree - while it doesn't really matter, using "Row" would be a better default (as that is the default tag for an outputted xml dataset).

richardkchapman commented:
@miguelvazq One question

miguelvazq commented:
@GordonSmith @richardkchapman Should be good now.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: a9547e3aa5bd1b15b5c26630a788b9c8e26ce6f6
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.642 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |10 |10 |0 |
|setup (thor) |10 |10 |0 |
|setup (roxie) |10 |10 |0 |
|test (hthor) |724 |724 |0 |
|test (thor) |619 |619 |0 |
|test (roxie) |747 |747 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
